File: cpp/fibonacci-iterative-recursive.cpp
#include <iostream>

int recursive_fib(int n) {
    if (n <= 1)
        return n;
    return recursive_fib(n-1) + recursive_fib(n-2);
}

int iterative_fib(int n) {
    if (n <= 1)
        return n;

    int prev = 1;
    int fib = 1;

    for (int i = 2; i < n; ++i) {
        int temp = fib;
        fib += prev;
        prev = temp;
    }
    return fib;
}

int main() {
    std::cout << recursive_fib(6) << std::endl;
    std::cout << recursive_fib(7) << std::endl;
    std::cout << recursive_fib(8) << std::endl;
    std::cout << iterative_fib(6) << std::endl;
    std::cout << iterative_fib(7) << std::endl;
    std::cout << iterative_fib(8) << std::endl;
    return 0;
}

File: py/fibonacci-iterative-recursive.py
def recursive_fib(n: int) -> int:
    """ Recursively calculates the n^th fibonacci number """
    if n <= 1:
        return n
    return recursive_fib(n-1) + recursive_fib(n-2)

def iterative_fib(n: int) -> int:
    """  """ 
    if n <= 1:
        return n
    prev = 1
    fib = 1
    for i in range(2, n):
        temp = fib
        fib += prev
        prev = temp
    return fib

print(recursive_fib(6))
print(recursive_fib(7))
print(recursive_fib(8))
print(iterative_fib(6))
print(iterative_fib(7))
print(iterative_fib(8))
